nltk/model/ngram.py

```diff
@@ -1,247 +1,72 @@
-
-        # we need to keep track of the number of word types we encounter
-        vocabulary = set()
-        for sent in train:
-            raw_ngrams = ngrams(sent, n, pad_left, pad_right, pad_symbol='')
-            for ngram in raw_ngrams:
-                self._ngrams.add(ngram)
-                context = tuple(ngram[:-1])
-                token = ngram[-1]
-                cfd[context][token] += 1
-                vocabulary.add(token)
-
-        # Unless number of bins is explicitly passed, we should use the number
-        # of word types encountered during training as the bins value.
-        # If right padding is on, this includes the padding symbol.
-        if 'bins' not in estimator_kwargs:
-            estimator_kwargs['bins'] = len(vocabulary)
-
-        self._model = ConditionalProbDist(cfd, estimator, **estimator_kwargs)
+class MLENgramModel(NgramModelI):
+    """An example of how to consume NgramCounter to create a language model.
 
-        # recursively construct the lower-order models
-        if not self.is_unigram_model:
-            self._backoff = NgramModel(n-1, train,
-                                        pad_left, pad_right,
-                                        estimator,
-                                        **estimator_kwargs)
-
-            self._backoff_alphas = dict()
-            # For each condition (or context)
-            for ctxt in cfd.conditions():
-                backoff_ctxt = ctxt[1:]
-                backoff_total_pr = 0.0
-                total_observed_pr = 0.0
-
-                # this is the subset of words that we OBSERVED following
-                # this context.
-                # i.e. Count(word | context) > 0
-                for word in self._words_following(ctxt, cfd):
-                    total_observed_pr += self.prob(word, ctxt)
-                    # we also need the total (n-1)-gram probability of
-                    # words observed in this n-gram context
-                    backoff_total_pr += self._backoff.prob(word, backoff_ctxt)
-
-                assert (0 <= total_observed_pr <= 1), total_observed_pr
-                # beta is the remaining probability weight after we factor out
-                # the probability of observed words.
-                # As a sanity check, both total_observed_pr and backoff_total_pr
-                # must be GE 0, since probabilities are never negative
-                beta = 1.0 - total_observed_pr
+    Currently untested.
+    """
 
-                # backoff total has to be less than one, otherwise we get
-                # an error when we try subtracting it from 1 in the denominator
-                assert (0 <= backoff_total_pr < 1), backoff_total_pr
-                alpha_ctxt = beta / (1.0 - backoff_total_pr)
+    def __init__(self, highest_order, unknown_cutoff, training_text,
+                 **ngram_counter_kwargs):
+        self.counter = NgramCounter(highest_order, unknown_cutoff, training_text,
```

Comment:
Same intuitive initialization here should be used (spill over from the refactoring the `NgramCounter.__init__`

```
 def __init__(self, training_text, highest_order, unknown_cutoff,
                  **ngram_counter_kwargs):
```